src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -401,9 +401,9 @@ You can use this event to run code on every page navigation, or only once ever:
 
 ### `astro:after-swap`
 
-An event that fires immediately after the new page replaces the old page. You can listen to this event on the `document`.
+An event that fires immediately after the new page replaces the old page. You can listen to this event on the `document` and trigger actions that will occur before the new page's scripts run. 
```

Comment:
```suggestion
An event that fires immediately after the new page replaces the old page. You can listen to this event on the `document` and trigger actions that will occur before the new page's DOM elements render and scripts run. 
```
What do you think about adding this too, while we're here? I've put this advice in the new tiny view transitions tutorial to explain why we're using it to check for dark mode to avoid a flash of light mode first. I think maybe it's even more helpful to say not just scripts? What do you think?

Reply:
Sure, that works, adjusted @sarah11918